src/main/java/com/mparticle/kits/GoogleAnalyticsFirebaseKit.java

```diff
@@ -209,6 +228,23 @@ PickyBundle getCommerceEventBundle(CommerceEvent commerceEvent) {
             Logger.info("Currency field required by Firebase was not set, defaulting to 'USD'");
             currency = "USD";
         }
+
+        // Google Analytics 4 introduces 2 new event types - add_shipping_info and add_payment_info
+        // each of these has an extra parameter that is optional
+        Map<String, List<String>> customFlags = commerceEvent.getCustomFlags();
+        if (customFlags != null && customFlags.containsKey(CF_GA4COMMERCE_EVENT_TYPE)) {
+            // TODO: Will get 0 throw an error if the array is empty?
+            String commerceEventType = customFlags.get(CF_GA4COMMERCE_EVENT_TYPE).get(0);
```

Comment:
@mchuangatmp @willpassidomo - can any android expert tell me about how if .get(0) on an empty array will just return null or if there is an elegant way of doing this to not throw an error if it does throw?

Reply:
it's gonna throw an `IndexOutOfBoundsException`, I probably would just wrap it around with a try/catch since it's Java